3741/1/top_gems.rb

```diff
@@ -0,0 +1,31 @@
+require 'yaml'
+
+require_relative 'lib/yaml_reader'
+require_relative 'lib/gem_data_reader'
+require_relative 'lib/gems_view'
+require_relative 'lib/my_option_parser'
+
+# main run class
+class TopGems
+  def self.render_table
+    top_gems = TopGems.new
+    top_gems.show
+  end
+
+  FILE_NAME = 'gems.yml'.freeze
+
+  def initialize
+    @options = MyOptionParser.parse_options
+  end
+
+  def read_file
+    YamlReader.read(@options[:file] || FILE_NAME, @options[:name])
```

Comment:
you should not call YamlReader if you have options[:file]

use guard clause here

Reply:
im not quite sure, but YamlReader need to call anyway with standart file name or rewrited in options